api/xcfun.h

```diff
@@ -14,9 +14,12 @@
 
 #pragma once
 
+#ifndef SWIG
```

Comment:
I tried removing this one, because I think we should keep this file as clean as possible from preprocessor macros and `ifdef`-s. Unfortunately the way SWIG works (in my limited experiments) makes it absolutely necessary.